### PR TITLE
Clarify that some exclude properties of physics query parameters are copied

### LINKED ISSUE #93895: PhysicsShapeQueryParameters3D's exclude array seems to be ignored in PhysicsDirectSpaceState3D intersect_shape func.
### Tested versions

v4.3.beta.custom_build [e2fc6d38c]

### System information

Godot v4.3.beta (e2fc6d38c) - Ubuntu 22.04.4 LTS 22.04 - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2060 (nvidia; 535.183.01) - Intel(R) Core(TM) i5-9400F CPU @ 2.90GHz (6 Threads)

### Explanation
I want to exclude a Node (self) from the collision test but it continues to be present. The MRP shows it all.

I have a RigidBody3D (a fish).
The fish has two CollisionShape3D nodes:
One is directly under the RigidBody3D (bumper) and is for collision with other fish.
The other is under an Area3D and is called 'ahead'. It is meant to look ahead and to the sides BUT NOT be part of the collisions with other fish.

I am using PhysicsDirectSpaceState3D and intersect_shape to get whatever 'ahead' may be overlapping. HOWEVER I do not want it to detect the fish itself, which it is doing.

I have tried various ways to exclude the RigidBody3D from the intersect process, but it just won't work.

Not sure if this is a bug or my own miscomprehension.

## Some code
With a RigidBody3D that has two CollisionShape3D nodes that overlap and this code:
```gdscript
q = PhysicsShapeQueryParameters3D.new()
q.exclude.append(self.get_rid()) # is not working later
q.collide_with_bodies = true
q.collision_mask = 2
q.shape = %senses/ahead.shape # my fish's "eyes"
q.margin = 0
```

```gdscript
# nf is an Array. It should be empty but is not. The (ahead) "eyes" see the main bumper collider.
# ss is the space state. 
nf = ss.intersect_shape(q, 2) 
```

### Steps to reproduce

Open the MRP, from Gitlab. Run the main scene. A fish will do a thing, but watch the output. The code is also commented. (issue.gd)

### Minimal reproduction project (MRP)

https://gitlab.com/dbat/fishy/-/tree/issue?ref_type=heads

### PR DESCRIPTION
**Pull request by ayanchavand**
Fixes #93895 
- Added the line` Note that [code]exclude[/code] is a replace-don't-edit type property and can only be modified by replacing the entire array.` to the PhysicsShapeQueryParameters3D.exclude docs

### CODE REVIEW

File: scene/resources/3d/primitive_meshes.cpp [hunk 1]
```diff
@@ -1011,7 +1011,7 @@ void CylinderMesh::create_mesh_array(Array &p_arr, float top_radius, float botto
 			u = i;
 			u /= radial_segments;
 
-			x = sin(u * Math_TAU);
+			x = safesin(u * Math_TAU);
```
Comment (AThousandShips) [inline scene/resources/3d/primitive_meshes.cpp:1014]: Unrelated, remove from this PR

**Review by AThousandShips — CHANGES_REQUESTED** [review]

**Comment by AThousandShips** [comment]
This should probably be done to *all* the cases of this, both 2D and 3D as well as the other parameters such as point and ray queries

**Comment by Mickeon** [comment]
[Your own PR](https://github.com/godotengine/godot/pull/85207) does this but it is done for every Packed Array property. To handles arrays like these would require a solution that allows Array properties to add the note automatically given a certain criteria

**Comment by AThousandShips** [comment]
That would be pretty complex as there's no way to detect it, so it'd be a matter of adding a new property to the XML class, not sure it's worth it, but might look into checking how many cases there are for this

**Comment by Mickeon** [comment]
Not this case specifically, but there are many times where information is repeated that may benefit from such a system, similar to the keywords.

**Comment by AThousandShips** [comment]
Absolutely, and I'll look at if there's enough cases to justify a system, but since this isn't automatable, and is likely not very common (i.e. cases that copy and *aren't* `Packed*Array`s which require a *different* message), it might not be worth it, but I'll take a look at adding an option to manually tag with `is_copied` or something to make the output text automated

**Comment by AThousandShips** [comment]
Please add to the other cases mentioned to get this to cover all the related cases, then this should be ready!

**Comment by AThousandShips** [comment]
You also merged your branch instead of using a force push, please use `git push --force` instead, and when you're done please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

**Comment by ayanchavand** [comment]
I think I accidentally created a branch from some other local branch instead of master, is it okay if I open new pr with the above changes?

**Comment by AThousandShips** [comment]
You can just rebase it, see the link, it shouldn't matter as long as you rebase on `master`

Also please make the final commit message something clear, like the PR title

**Review by AThousandShips — APPROVED** [review]

**Review by Mickeon — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**